CHANGELOG.md

```diff
@@ -7,6 +7,7 @@
 * Added support for querying inverse relationships (#2904).
 * Moved inverse relationships out of beta stage.
 * Added `Realm.getDefaultConfiguration()` (#4725).
+* Added `RealmObject.getConfiguration(RealmModel)` (#4720).
```

Comment:
We definitely need to duplicate the methods across the `RealmObject` base class and the static methods. People should be able to use both interchangeably.

Adding it to the `RealmObject` base class could be potentially breaking though if people have their own `getConfiguration()` method. For that reason alone, I'm not sure it is worth sneaking this change into a minor release, and we should probably wait for 4.0

Also, I'm still trying to convince myself that this is truly useful compared to exposing `RealmObject.getRealm()`.

```
public foo(Person p) {
  Realm realm = Realm.getInstance(p.getConfiguration());
  realm.executeTransaction({ p.deleteFromRealm(); });
  realm.close();
}

vs.

public foo(Person p) {
  Realm realm = p.getRealm();
  realm.executeTransaction({ p.deleteFromRealm(); });
}
```

I understand the argument that exposing a `getRealm()` method makes it harder to reason about the Realm lifecycle + we don't have to battle the `DynamicRealm/Realm` issues, but it still feels wrong to me somehow. It is like requiring people to always do one extra step of work, only because we want them to not accidentally call `realm.close()`. What are your thoughts on this @realm/java?